cache.go

```diff
@@ -318,8 +432,213 @@ func (c *GoBuildCache) ToRunOption(distroKey string, opts ...GoBuildCacheOption)
 	})
 }
 
-// BazelCache sets up a cache for bazel builds.
-//
+// CargoBuildCache is a cache for Rust/Cargo build artifacts.
+// It uses sccache to speed up Rust compilation by caching build artifacts.
+type CargoBuildCache struct {
```

Comment:
Maybe let's call this `CargoSCCache` (and `cargosccache` in the yaml... or something better, just a suggestion) since the implementation is tied to sccache and if the tooling is ever different we'd need a different cache key anyway.